Add the missing `list_merge='combine'` option (#75110)
ansible/ansible: docs/docsite/rst/user_guide/playbooks_filters.rst

    b: patch
    c: default

> If ``list_merge='combine'`` (SHOULD be the default), arrays from the right hash will "combine" with the ones in the left hash in exactly the same way keys in hashes "combine" (i.e. this option essentially treats arrays as hashes with numeric keys)::

Reviewer: 'SHOULD be' the default seems rather subjective since the lists need to be in a predictable order.

```suggestion
If ``list_merge='combine'`` (not the default), arrays from the right hash will "combine" with the ones in the left hash in exactly the same way keys in hashes "combine" (i.e. this option essentially treats arrays as hashes with numeric keys)::
```

Author: I agree.  I just couldn't resist... 😈

      - 1
      - 1
      - 2
      - default_key: default_value
      - 3
      - 4
      - 4
      - patch_key: patch_value

The ``list_merge='combine'`` option shows its true strength when combined with ``recusive=True``, as it traverse complex data structures fully, combining both hash keys and array items::

    {{ default | combine(patch, recursive=True, list_merge='combine') }}

This would result in::

    a:
      a':
        x: default_value
        y: patch_value
        z: patch_value
        list:
          - patch_value
    b:
      - 3
      - 4
      - 4
      - default_key: default_value
        patch_key: patch_value

>       - 3

Reviewer: Looks like an extra?
```suggestion
```

Reviewer: Oops, I had a copy-pasta fail :-)
